test/functional/vimscript/system_spec.lua

```diff
@@ -190,6 +190,27 @@ describe('system()', function()
       eq(iswin() and '??\n' or 'ああ\n', eval([[system('Write-Output "ああ"')]]))
     end)
 
+    it('check powershell filter and redirection (#16271)', function()
```

Comment:
since there is no unix test here, let's add it to the windows-only tests here:

https://github.com/neovim/neovim/blob/e420cd6c67a0aad5b5dcb40748f733876e66a2c3/test/functional/vimscript/system_spec.lua#L128

Reply:
So, the test should be run for Unix, too. So I moved it out of `if iswin()`.